mod_wtjshoppingfavorites/src/Dispatcher/Dispatcher.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * WT JoomShopping Favorites is an alternative wish list (favorite products) for JoomShopping based on cookies.
+ * @package     WT JoomShopping Favorite
+ * @version     2.0.2
+ * @Author      Sergey Tolkachyov, https://web-tolk.ru
+ * @copyright   Copyright (C) 2024 Sergey Tolkachyov
+ * @license     GNU/GPL 3.0
+ * @since       1.0.0
+ * @link        https://web-tolk.ru/en/dev/joomshopping/wt-joomshopping-favorite.html
+ */
+namespace Joomla\Module\Wtjshoppingfavorites\Site\Dispatcher;
+
+\defined('JPATH_PLATFORM') or die;
+
+use Joomla\CMS\Application\CMSApplicationInterface;
+use Joomla\CMS\Dispatcher\AbstractModuleDispatcher;
+use Joomla\CMS\Extension\ModuleInterface;
+use Joomla\CMS\Plugin\PluginHelper;
+use Joomla\CMS\WebAsset\WebAssetManager;
+use Joomla\Input\Input;
+use Joomla\Registry\Registry;
+
+class Dispatcher extends AbstractModuleDispatcher
+{
+    /**
+     * The module extension. Used to fetch the module helper.
+     *
+     * @var   ModuleInterface|null
+     * @since 1.0.0
+     */
+    private $moduleHelper;
+
+    public function __construct(\stdClass $module, CMSApplicationInterface $app, Input $input)
+    {
+        parent::__construct($module, $app, $input);
+        $this->moduleHelper = $this->app->bootModule('mod_wtjshoppingfavorites', 'site')->getHelper('WtjshoppingfavoritesHelper');
+    }
+
+    /**
+     * Returns the layout data.
+     *
+     * @return  array
+     *
+     * @since   1.0.0
+     */
+    protected function getLayoutData(): array
+    {
+        $data = parent::getLayoutData();
+        $app = $this->getApplication();
+
+        $data['moduleId'] = $data['module']->id;
+        if (!empty($data['params']->get('moduleclass_sfx','')))
+        {
+            $data['moduleclass_sfx'] = htmlspecialchars($data['params']->get('moduleclass_sfx',''), ENT_COMPAT, 'UTF-8');
+        }
+        $data['product_ids'] = $this->moduleHelper->getProductList($data['params'], $app);
+
+
+        $itemid = $this->moduleHelper->getItemid('com_jshopping', 'wtjshoppingfavorites');
+        if (empty($itemid) || is_null($itemid))
```

Comment:
empty проверяет на null тоже https://www.php.net/manual/en/function.empty

```suggestion
        if (empty($itemid))
```